### PR TITLE
WorkdaysBetween: Convert to full template

### PR DESCRIPTION
**Pull request by rasikapohankar**
###### What does your Pull Request do (check all that apply)?

- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [x] Enhancement: **`WorkdaysBetween: Convert to full template`**

###### Description of changes

Updates WorkdaysBetween to use the full template. Re: #2767 

###### People to notify (@mention interested parties)
@GuiltyDolphin @moollaza 

---

Instant Answer Page: https://duck.co/ia/view/workdays_between

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mgarriott

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by GuiltyDolphin** [comment]
@rasikapohankar Meeerge conflicts!

Make sure to only use `re(...)` when you want to do a regex test!

**Comment by rasikapohankar** [comment]
@GuiltyDolphin Yes..the file changed after I had started editing it. Now I should take those changes and make my changes in the new file?

**Comment by GuiltyDolphin** [comment]
@rasikapohankar You can probably just overwrite those changes with yours, but make sure to use `Test::Deep` and the `re` wrapper!

**Comment by rasikapohankar** [comment]
@GuiltyDolphin I have used `Test::Deep` and `re` inside `build_structured_answer`. Is it also necessary to use `re` in the call to `build_test`? If I use it in the call to `build_test` it says, 
`# Using Regexp on $data->{"structured_answer"}{"data"}{"subtitle"}                                                                                                  
    #    got : 'Workdays between 10 Jan 2015 - 20 Jan 2015'                                                                                                             
    # expect : (?^:Workdays between Test::Deep::Regexp=HASH(0x17bb928) - Test::Deep::Regexp=HASH(0x17bba48))`.
How can I overwrite the changes?

**Comment by GuiltyDolphin** [comment]
@rasikapohankar You might be able to use `git rebase --theirs`, or failing that, just do a rebase and always favor your changes.

As long as any regexes in the response are wrapped in `re` then it's all good!

**Comment by rasikapohankar** [comment]
@GuiltyDolphin Thanks!! :+1: 
Is this okay now?

File: t/WorkdaysBetween.t [hunk 1]
```diff
@@ -11,225 +11,140 @@ use Test::MockTime qw( :all );
...
+     'Workdays between 01/06/2014 01/10/2014'           => build_test('06 Jan 2014', '10 Jan 2014', 5),
+     'Workdays between 01/06/2014 01/10/2014 inclusive' => build_test('06 Jan 2014', '10 Jan 2014', 5),
+
+     # Ending date first
+     'Workdays between 01/10/2014 01/06/2014'           => build_test('06 Jan 2014', '10 Jan 2014', 5),
+     'Workdays between 01/10/2014 01/06/2014 inclusive' => build_test('06 Jan 2014', '10 Jan 2014', 5),
+
+     # Ending date on a weekend
+     'Workdays between 01/06/2014 01/12/2014'           => build_test('06 Jan 2014', '12 Jan 2014', 5),
+     'Workdays between 01/06/2014 01/12/2014 inclusive' => build_test('06 Jan 2014', '12 Jan 2014', 5),
```
Comment (GuiltyDolphin) [inline t/WorkdaysBetween.t]: @rasikapohankar We should switch any ambiguous formats to using ISO8601 (`YYYY-MM-DD` e.g, `2014-01-06`); we're testing how the Goodie handles the dates, not how it parses them (which is handled by the Role).

File: t/WorkdaysBetween.t [hunk 2]
```diff
@@ -11,225 +11,140 @@ use Test::MockTime qw( :all );
...
+     'Workdays between 01/06/2014 01/13/2014'           => build_test('06 Jan 2014', '13 Jan 2014', 6),
+     'Workdays between 01/06/2014 01/13/2014 inclusive' => build_test('06 Jan 2014', '13 Jan 2014', 6),
+
+     # Standard work week
+     'Workdays between 01/06/2014 01/10/2014'           => build_test('06 Jan 2014', '10 Jan 2014', 5),
+     'Workdays between 01/06/2014 01/10/2014 inclusive' => build_test('06 Jan 2014', '10 Jan 2014', 5),
+
+     # Ending date first
+     'Workdays between 01/10/2014 01/06/2014'           => build_test('06 Jan 2014', '10 Jan 2014', 5),
+     'Workdays between 01/10/2014 01/06/2014 inclusive' => build_test('06 Jan 2014', '10 Jan 2014', 5),
```
Comment (GuiltyDolphin) [inline t/WorkdaysBetween.t]: We should be able to pull some of these cases out into vars, eg., `my @jan_6_to_10 = ('06 Jan 2014', '10 Jan 2014', 5);` then use it as `... => build_test(@jan_6_to_10)`. We should do this for any tests with repeated results.

**Comment by rasikapohankar** [comment]
@GuiltyDolphin Have made suggested the changes.

File: t/WorkdaysBetween.t [hunk 3]
```diff
@@ -11,225 +11,153 @@ use Test::MockTime qw( :all );
...
+    'Workdays between 2014-01-13 2014-01-06'           => build_test(@jan_6_to_13),
+    'Workdays between 2014-01-13 2014-01-06 inclusive' => build_test(@jan_6_to_13),
+
+     # Including the weekend
+    'Workdays between 2014-01-06 2014-01-13'           => build_test(@jan_6_to_13),
+    'Workdays between 2014-01-06 2014-01-13 inclusive' => build_test(@jan_6_to_13),
+
+     # Standard work week
+    'Workdays between 01/06/2014 01/10/2014'           => build_test(@jan_6_to_10),
+    'Workdays between 01/06/2014 01/10/2014 inclusive' => build_test(@jan_6_to_10),
```
Comment (GuiltyDolphin) [inline t/WorkdaysBetween.t]: As mentioned before, lets update the `/` dates to ISO format.

File: t/WorkdaysBetween.t [hunk 4]
```diff
@@ -11,225 +11,153 @@ use Test::MockTime qw( :all );
...
+    'Workdays between 01-01-2014 01-01-2015 inclusive' => build_test(@jan_2014_to_2015),
+
+     # Single digit days and months
+    'Workdays between 2014-01-06 2014-01-10'           => build_test(@jan_6_to_10),
+    'Workdays between 2014-01-06 2014-01-10 inclusive' => build_test(@jan_6_to_10),
+
+     # Month and Date are backwards
+    'Workdays between 16/06/2014 20/06/2014' => build_test(@jun_16_to_20),
+    'Workdays between 2014-06-05 2014-06-20'  => build_test(@june_5_to_20),
+    'Workdays between 2014-06-20 2014-06-05'  => build_test(@june_5_to_20),
```
Comment (GuiltyDolphin) [inline t/WorkdaysBetween.t]: We can remove these three, they don't test the Goodie only the Role.

File: t/WorkdaysBetween.t [hunk 5]
```diff
@@ -11,225 +11,153 @@ use Test::MockTime qw( :all );
...
+    'Workdays between jan 6 2014 jan 10 2014 inclusive' => build_test(@jan_6_to_10),
+
+     # Single digit days and months - Dash format
+    'Workdays between 1-6-2014 1-10-2014'           => build_test(@jan_6_to_10),
+    'Workdays between 1-6-2014 1-10-2014 inclusive' => build_test(@jan_6_to_10),
+
+     # Month and Date are backwards - Dash format
+    'Workdays between 16-06-2014 20-06-2014' => build_test(@jun_16_to_20),
+    'Workdays between 5-06-2014 20-06-2014'  => build_test(@june_5_to_20),
+    'Workdays between 20-06-2014 5-06-2014'  => build_test(@june_5_to_20),
```
Comment (GuiltyDolphin) [inline t/WorkdaysBetween.t]: We can remove these ones.

File: t/WorkdaysBetween.t [hunk 6]
```diff
@@ -11,225 +11,153 @@ use Test::MockTime qw( :all );
...
+
+     #same day on a weekend
+    'Workdays between jan 4, 2014 jan 4, 2014'           => build_test(@jan_4_samedaywknd),
+    'Workdays between jan 4, 2014 jan 4, 2014 inclusive' => build_test(@jan_4_samedaywknd),
+
+     # Same day
+    'Workdays between jan 3, 2014 jan 3, 2014'           => build_test(@jan_3_sameday),
+    'Workdays between jan 3, 2014 jan 3, 2014 inclusive' => build_test(@jan_3_sameday),
+
+     # Unambiguous date format with comma separator
```
Comment (GuiltyDolphin) [inline t/WorkdaysBetween.t]: We can delete all the tests from here down to `Workdays in a year`

File: t/WorkdaysBetween.t [hunk 7]
```diff
@@ -11,225 +11,153 @@ use Test::MockTime qw( :all );
...
-            input     => ignore(),
-            operation => 'Workdays between',
-            result    => re(qr/[1-9][0-9]/),
-        }
-    ),
+    [qw(DDG::Goodie::WorkdaysBetween)],       
+
+    'Workdays between 01/31/2000 01/31/2001'               => build_test(@jan_31_2000_to_2001),
+    'Workdays between 01/31/2000 and 01/31/2001 inclusive' => build_test(@jan_31_2000_to_2001),
+    'Workdays between 01/31/2000 01/31/2001 inclusive'     => build_test(@jan_31_2000_to_2001),
```
Comment (GuiltyDolphin) [inline t/WorkdaysBetween.t]: Lets update these to use ISO8601.

**Comment by rasikapohankar** [comment]
@GuiltyDolphin Can you please check?

**Comment by GuiltyDolphin** [comment]
@rasikapohankar Thanks!

**Pull request merged**